CRM/Core/Task.php

```diff
@@ -165,6 +165,51 @@ public static function getTask($value) {
     ];
   }
 
+  /**
+   * Filter tasks based on the permission key, if available.
+   *
+   * @param array $tasks
+   * @param bool $hasEditContactPermission
+   *   Does the user have permission to edit the given contact. Required where
+   *   permission to edit the user is required in conjunction with permission
+   *   to do the task.
+   *
+   * @return array
+   */
+  protected static function getFilteredTasks(array $tasks, bool $hasEditContactPermission): array {
```

Comment:
Maybe `getTasksFilteredByPermission()` to be more explicit/clear what this does?

Reply:
Yep - updated to that (& the other function to match)